addons/mail/static/src/messaging/entity/attachment/attachment.js

```diff
@@ -0,0 +1,305 @@
+         */
+        static AllFromFilename(filename) {
+            const allAttachments = this.all;
+            return allAttachments.filter(attachment => attachment.filename === filename);
+        }
+
+        /**
+         * @param {string} filename
+         * @returns {mail.messaging.entity.Attachment|undefined}
+         */
+        static temporaryFromFilename(filename) {
+            const attachments = this.AllFromFilename(filename);
+            return attachments.find(attachment => attachment.isTemporary);
+        }
+
+        /**
+         * View provided attachment(s), with given attachment initially. Prompts
+         * the attachment viewer.
+         *
+         * @param {Object} param0
+         * @param {mail.messaging.entity.Attachment} [param0.attachment]
+         * @param {mail.messaging.entity.Attachment>[]} param0.attachments
+         * @returns {string|undefined} unique id of open dialog, if open
+         */
+        static view({ attachment, attachments }) {
+            if (!attachments) {
+                return;
+            }
+            if (!attachment) {
+                attachment = attachments[0];
+            }
+            if (!attachments.includes(attachment)) {
+                return;
+            }
+            this.env.entities.Dialog.open('AttachmentViewer', { attachment, attachments });
+        }
+
+        /**
+         * @returns {string}
+         */
+        get defaultSource() {
+            const fileType = this.fileType;
+            if (fileType === 'image') {
+                return `/web/image/${this.id}?unique=1&amp;signature=${this.checksum}&amp;model=ir.attachment`;
+            }
+            if (fileType === 'application/pdf') {
+                return `/web/static/lib/pdfjs/web/viewer.html?file=/web/content/${this.id}?model%3Dir.attachment`;
+            }
+            if (fileType && fileType.indexOf('text') !== -1) {
+                return `/web/content/${this.id}?model%3Dir.attachment`;
+            }
+            if (fileType === 'youtu') {
+                if (fileType !== 'youtu') {
+                    return undefined;
+                }
+                const urlArr = this.url.split('/');
+                let token = urlArr[urlArr.length - 1];
+                if (token.indexOf('watch') !== -1) {
+                    token = token.split('v=')[1];
+                    const amp = token.indexOf('&');
+                    if (amp !== -1) {
+                        token = token.substring(0, amp);
+                    }
+                }
+                return `https://www.youtube.com/embed/${token}`;
+            }
+            if (fileType === 'video') {
+                return `/web/image/${this.id}?model=ir.attachment`;
+            }
+            return undefined;
+        }
+
+        /**
+         * @returns {string}
+         */
+        get displayName() {
+            return this.name || this.filename;
+        }
+
+        /**
+         * @returns {string}
```

Comment:
This can probably return something else than string, at least undefined. Not sure if intended.